services/intersection-api/api/src/main/java/us/dot/its/jpo/ode/api/accessors/events/aggregations/bsmmessagecountprogression/BsmMessageCountProgressionEventAggregationRepositoryImpl.java

```diff
@@ -0,0 +1,112 @@
+package us.dot.its.jpo.ode.api.accessors.events.aggregations.bsmmessagecountprogression;
+
+import javax.annotation.Nullable;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Sort;
+import org.springframework.data.mongodb.core.MongoTemplate;
+import org.springframework.data.mongodb.core.query.Criteria;
+import org.springframework.data.mongodb.core.query.Query;
+import org.springframework.stereotype.Component;
+
+import org.springframework.data.domain.Pageable;
+
+import us.dot.its.jpo.conflictmonitor.monitor.models.events.BsmMessageCountProgressionEventAggregation;
+import us.dot.its.jpo.ode.api.accessors.IntersectionCriteria;
+import us.dot.its.jpo.ode.api.accessors.PageableQuery;
+
+@Component
+public class BsmMessageCountProgressionEventAggregationRepositoryImpl
+        implements BsmMessageCountProgressionEventAggregationRepository, PageableQuery {
+
+    private final MongoTemplate mongoTemplate;
+
+    private final String collectionName = "CmBsmMessageCountProgressionEventAggregation";
+    private final String DATE_FIELD = "eventGeneratedAt";
+    private final String INTERSECTION_ID_FIELD = "intersectionID";
+
+    @Autowired
+    public BsmMessageCountProgressionEventAggregationRepositoryImpl(MongoTemplate mongoTemplate) {
+        this.mongoTemplate = mongoTemplate;
+    }
+
+    /**
+     * Get a page representing the count of data for a given intersectionID,
+     * startTime, and endTime
+     *
+     * @param intersectionID the intersection ID to query by, if null will not be
+     *                       applied
+     * @param startTime      the start time to query by, if null will not be applied
+     * @param endTime        the end time to query by, if null will not be applied
+     * @param pageable       the pageable object to use for pagination
+     * @return the paginated data that matches the given criteria
+     */
+    public long count(
+            Integer intersectionID,
+            Long startTime,
+            Long endTime,
+            @Nullable Pageable pageable) {
```

Comment:
We removed pageable from the count queries, this affects the parameters, the comment, and the interface parameter def